include/picongpu/plugins/transitionRadiation/TransitionRadiation.hpp

```diff
@@ -449,6 +528,177 @@ namespace picongpu
+
+                    mesh.setUnitDimension(std::map<::openPMD::UnitDimension, double>{
+                        {::openPMD::UnitDimension::L, 2.0},
+                        {::openPMD::UnitDimension::M, 1.0},
+                        {::openPMD::UnitDimension::T, -1.0}});
+
+                    auto transitionRadiation = mesh[::openPMD::RecordComponent::SCALAR];
+                    transitionRadiation.resetDataset(dataset);
+
+                    transitionRadiation.setUnitSI(
+                        SI::ELECTRON_CHARGE_SI * SI::ELECTRON_CHARGE_SI
+                        * (1.0 / (4 * PI * SI::EPS0_SI * PI * PI * SI::SPEED_OF_LIGHT_SI)));
+
+                    auto span = transitionRadiation.storeChunk<float_X>(offset, extent);
+                    auto spanBuffer = span.currentBuffer();
+
+                    for(unsigned int index_direction = 0; index_direction < transitionRadiation::parameters::nObserver;
+                        ++index_direction)
+                    {
+                        // theta
+                        const int i = index_direction / parameters::nPhi;
+                        // phi
+                        const int j = index_direction % parameters::nPhi;
+
+                        for(unsigned int k = 0; k < transitionRadiation::frequencies::nOmega; ++k)
+                        {
+                            const int index = (k * parameters::nPhi + j) * parameters::nTheta + i;
+                            spanBuffer[index] = static_cast<float_X>(
+                                theTransRad[index_direction * transitionRadiation::frequencies::nOmega + k]);
+                        }
+                    }
+
+                    // Omega axis
+                    ::openPMD::Extent extentOmega
+                        = {static_cast<unsigned long int>(transitionRadiation::frequencies::nOmega), 1, 1};
+                    ::openPMD::Offset offsetOmega = {0, 0, 0};
+                    ::openPMD::Datatype datatypeOmega = ::openPMD::determineDatatype<float_X>();
+                    ::openPMD::Dataset datasetOmega{datatypeOmega, extentOmega};
+
+                    auto meshOmega = iteration.meshes["detector omega"];
+
+                    meshOmega.setAxisLabels(std::vector<std::string>{"omega", "", ""});
+                    meshOmega.setDataOrder(::openPMD::Mesh::DataOrder::C);
+                    meshOmega.setGridUnitSI(1);
+                    meshOmega.setGridSpacing(std::vector<double>{1, 1, 1});
+                    meshOmega.setGeometry(::openPMD::Mesh::Geometry::cartesian); // set be default
+
+                    meshOmega.setUnitDimension(
+                        std::map<::openPMD::UnitDimension, double>{{::openPMD::UnitDimension::T, -1.0}});
+
+                    auto omega = meshOmega[::openPMD::RecordComponent::SCALAR];
+                    omega.resetDataset(datasetOmega);
+
+                    omega.setUnitSI(1.0);
```

Comment:
shouldn't that be `1/UNIT_TIME`?

Reply:
The frequencies given by `transitionRadiation::frequencies::FreqFunctor` are not normalized, but in SI. So it should just be 1.0